src/lib.rs

```diff
@@ -110,7 +126,7 @@ pub trait Core: Send + Sync {
     /// call [Core::complete_workflow_task] for those workflows until they are done. At that point,
     /// the lang SDK can end the process, or drop the [Core] instance, which will close the
     /// connection.
```

Comment:
This should be updated to indicate that now that it's async it returns when shutdown is actually all the way done

Reply:
And, come to think of it, should return a result now and propagate errors but just make a task for that and we can do it after this. Shouldn't take long.

Reply:
Sounds good, slightly updated the comment.